File: cdm/cpp/scenario/SEAction.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "stdafx.h"
#include "scenario/SEAction.h"
#include "substance/SESubstanceManager.h"

#include "scenario/SEAdvanceTime.h"
#include "scenario/SESerializeState.h"


SEAction::SEAction() : Loggable()
{
  m_Comment="";
}

SEAction::~SEAction()
{
  Clear();
}

void SEAction::Clear()
{
  m_Comment="";
}

std::string SEAction::GetComment() const
{
  return m_Comment;
}
void SEAction::SetComment(const std::string& comment)
{
  m_Comment = comment;
}
bool SEAction::HasComment() const
{
  return m_Comment.empty()?false:true;
}
void SEAction::InvalidateComment()
{
  m_Comment = "";
}

File: src/csharp/pulse/cdm/engine/SEAction.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

using System;
using System.Linq;

namespace Pulse.CDM
{
  public enum eMergeType : int
  {
    Append = 0,
    Replace
  }
  
  public enum eAppliedRespiratoryCycle : int
  {
    Expiratory = 0,
    Inspiratory,
    Instantaneous
  }
  public abstract class SEAction
  {
    protected string comment;

    public SEAction()
    {
      comment = null;
    }

    public void Copy(SEAction other)
    {
      Clear();
      comment = other.comment;
    }

    public virtual void Clear()
    {
      comment = null;
    }

    public bool HasComment()
    {
      return !string.IsNullOrEmpty(comment);
    }
    public string GetComment()
    {
      return this.comment;
    }
    public void SetComment(string comment)
    {
      this.comment = comment;
    }
    public void InvalidateComment()
    {
      this.comment = null;
    }

    public abstract bool IsValid();

    public static string PrettyPrint(string msg)
    {
      return Logger.PrettyPrint(msg, Logger.ePrettyPrintType.Action);
    }
  }
}
